### ISSUE TITLE
Moderator (DB user) is not automatically created

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#3940]
When creating infrastructure and doing automatic DB setup, the moderator user is something I did not get to. AFAIK it does not exist on prod. 

Given our conversations for an http server in #3865 ; I recommend we instead focus on getting that up and running and providing endpoints to return information to moderator users. It's a different mechanism to achieve the same, but a more useful/flexible result that will be easier to use. I think that would be a better focus/effort, if there is general agreement we can potentially close this issue with that consensus/plan.

There is a question of how to proceed, marking as discussion until we determine how to move forward.

**Comment by prastle** [comment]
@DanVanAtta  actually it does still exist I believe.  You just have to switch to it once ssh in.  Used it last week.  Perhaps I misunderstand what you are saying.

**Comment by DanVanAtta** [comment]
@prastle thank you for taking a look and checking, that clarifies the problem which I mistated. The creation of the DB moderator user is no longer in the automated setup. In my mind that often equates to not existing. You should find in the old prerelease server, and the newly created one, that the moderator user is not there. This is only going to impact developer testing or a moderator that wants to demo a prerelease feature that needs SQL access (never going to impact prod unless disaster strikes and DB data is completely wiped out).

**Comment by prastle** [comment]
@DanVanAtta  wrong... If you remove it none of @ssoloff hard work on the db checks for moderator will work.  Hopefully I am incorrect on this but please check with steven first

**Comment by prastle** [comment]
example who is user x

**Comment by DanVanAtta** [comment]
@prastle it's not necessarily removal, it's more that the user creation is not part of the automation that sets things up from scratch. So if we lose the current server and recreate it with the automation, it would be missing (and hence why prerelease is missing it). 

The moderator user was set up before infrastructure was created, and highlights a problem solved by infrastructure. Before we had to know everything that was done from one server to another, and when creating a new one apply it all. (Naturally the knowledge degrades, when I was handed role of lead sys admin in charge of all servers, there was basically nothing to document how things were set up. If we had a full system crash there was a time we may not have been able to recreate everything)

Long story short, we need the DB user creation commands to be entered into the infrastructure scripts to resolve this issue, this way we do not have to run those commands on each system by hand when setting a DB up from scratch.

**Comment by prastle** [comment]
I understand your point but please hear mine for a second.  It took many years for us to finally to achieve where we are today and a working database that can be referenced by those with ssh.   I guess all I am saying is make sure we can acess what has finally been saved before ya remove it. so it can be reinstalled by infra.

**Comment by ssoloff** [comment]
> The moderator user was set up before infrastructure was created, and highlights a problem solved by infrastructure.

IIRC, the `moderator` user was not added to solve any infrastructure issues but was simply a "safe" (i.e. no sudo privileges) account from which lobby mods could fire up `psql` and run some queries against the lobby DB (using a corresponding `moderator` Postgres user that only had read access to the `ta_users` DB).

It was intended as a quick solution to address issues raised by lobby mods (e.g. cross-referencing lobby users by MAC to help identify trolls).  I don't remember where this was discussed, but I believe the agreement was to provide this read-only DB access while we worked on adding equivalent functionality to the lobby client or a dedicated web front-end.

**Comment by prastle** [comment]
Yup you covered it completely ty @ssoloff

**Comment by prastle** [comment]
Problem is that only those that can ssh to the server can then switch to use moderator.. Thus us..  Either way tho at least it can be viewed easily again if we ever help them get ssh

**Comment by DanVanAtta** [comment]
The issue / problem is not clear. The moderator user was added by hand, when we re-create the DB, it must be re-created by hand again. The task here is to instead of running those commands by hand, to commit them to our DB cookbook so it is installed automatically. That is a pro-active step to add it to automation that is above the fact that at one point we manually ran the user create command on production.

**Comment by DanVanAtta** [comment]
Moderators for some time AFAIK do not have access. They would need to create a SSH key. I'm interested in following up on exactly that for access to run 'ban-ip' as well. Without even moderator SSH access, a DB user for it does not make sense. I'm going to close this tracking ticket for now as we will follow up on this topic as needed in other threads. Notably I'll be working with mods for SSH access, if we get that far then we can revisit how to query DB. With some luck at that time we'll have made advances in our moderator tooling and would not need SSH access anyways (in other words, by then we might have a mod-only GUI toolbox that can provide this for us instead of using command line tools)

**Comment by prastle** [comment]
makes sense since they have no ssh
